src/databricks/sql/auth/retry.py

```diff
@@ -222,7 +224,7 @@ def new(self, **urllib3_incremented_counters: typing.Any) -> Retry:
         urllib3_init_params.update(**urllib3_incremented_counters)
 
         # Include urllib3's current state in our __init__ params
-        databricks_init_params["urllib3_kwargs"].update(**urllib3_init_params)  # type: ignore
+        databricks_init_params["urllib3_kwargs"] = urllib3_init_params.copy()
```

Comment:
This modification actually changes the behavior of the code (replacing the whole dict is not the same as adding new/updating existing keys - even if for now it works because initial dict is empty). Let's revert to updating